java/fury-core/src/main/java/org/apache/fury/serializer/Serializers.java

```diff
@@ -364,14 +364,13 @@ public BigIntegerSerializer(Fury fury) {
     @Override
     public void write(MemoryBuffer buffer, BigInteger value) {
       final byte[] bytes = value.toByteArray();
-      Preconditions.checkArgument(bytes.length <= 16);
-      buffer.writeByte((byte) bytes.length);
+      buffer.writePositiveVarInt((byte) bytes.length);
```

Comment:
Directly pass in `bytes.length` to `writePositiveVarInt`. Why do you need to convert it to byte? This will cause truncation.

Reply:
Nice catch, fixed